src/nvim/tui/tui.c

```diff
@@ -792,11 +792,13 @@ static void cursor_goto(TUIData *tui, int row, int col)
   if (grid->row == -1) {
     goto safe_move;
   }
-  if (0 == col ? col != grid->col :
-      row != grid->row ? false :
-      1 == col ? 2 < grid->col && cheap_to_print(tui, grid->row, 0, col) :
-      2 == col ? 5 < grid->col && cheap_to_print(tui, grid->row, 0, col) :
-      false) {
+  if (0 == col ? col != grid->col
+               : row != grid->row ? false
+                                  : 1 == col ? 2 < grid->col
+      && cheap_to_print(tui, grid->row, 0, col)
```

Comment:
This a little misleading. Parens for `(2 < grid->col && cheap_to_print(tui, grid->row, 0, col))`  would be good anyway, and maybe helps?